### PR TITLE
Discovery Provision with parameters bug #1572947 automated

### PR DESCRIPTION
**Pull request by jyejare**
Bug 1572947 is automated.

### CODE REVIEW

**Comment by jyejare** [comment]
Without Patch:
```
$ py.test tests/foreman/cli/test_discoveredhost.py -k test_positive_provision_pxe_host_with_parameters
2019-08-01 16:42:46 - conftest - DEBUG - Registering custom pytest_configure
================================================ test session starts =================================================
platform linux -- Python 3.6.8, pytest-4.6.3, py-1.7.0, pluggy-0.12.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/jitendrayejare/Desktop/RedHat/RoboTelloNew/robottelo
plugins: forked-1.0.2, services-1.3.1, xdist-1.26.1, mock-1.10.4, cov-2.6.1
collecting ... 2019-08-01 16:43:10 - conftest - DEBUG - Collected 25 test cases

collected 25 items / 24 deselected / 1 selected                                                                      

tests/foreman/cli/test_discoveredhost.py s                                                                     [100%]
=============================== 1 skipped, 24 deselected in 113.38 seconds ===============================
```

With Patch (https://bugzilla.redhat.com/show_bug.cgi?id=1572947#c11) :
```
$ py.test tests/foreman/cli/test_discoveredhost.py -k test_positive_provision_pxe_host_with_parameters
2019-08-01 16:42:46 - conftest - DEBUG - Registering custom pytest_configure
================================================ test session starts =================================================
platform linux -- Python 3.6.8, pytest-4.6.3, py-1.7.0, pluggy-0.12.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/jitendrayejare/Desktop/RedHat/RoboTelloNew/robottelo
plugins: forked-1.0.2, services-1.3.1, xdist-1.26.1, mock-1.10.4, cov-2.6.1
collecting ... 2019-08-01 16:43:10 - conftest - DEBUG - Collected 25 test cases

collected 25 items / 24 deselected / 1 selected                                                                      

tests/foreman/cli/test_discoveredhost.py .                                                                     [100%]
=============================== 1 passed, 24 deselected in 439.16 seconds ================================
```

File: tests/foreman/cli/test_discoveredhost.py [hunk 1]
```diff
@@ -836,3 +836,63 @@ def test_positive_update_discover_hostname_settings(self):
...
+               after discovery
+            3. PXE Boot the VM (from Network)
+            4. Wait for Host to be discovered by Satellite
+            5. Provision the discovered host with host parameters
+
+        :expectedresults:
+            1. Ensure host appeared in Discovered Hosts on satellite
+            2. Ensure the host is provisioned with correct and all the host parameters
+               given during provisioning
+            3. Ensure the entry from discovered host list disappeared
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_discoveredhost.py]: As you mentioned above-Discovered Host List, but you check on info command

File: tests/foreman/cli/test_discoveredhost.py [hunk 2]
```diff
@@ -836,3 +836,63 @@ def test_positive_update_discover_hostname_settings(self):
...
+            '{0}={1}, {2}={3}'.format(param1_key, param1_value, param2_key, param2_value)]
+        # fixme: assertion #1
+        if not self.configured_env:
+            self.__class__.configured_env = configure_env_for_provision(
+                org=self.org, loc=self.loc)
+        with LibvirtGuest() as pxe_host:
+            hostname = pxe_host.guest_name
+            # fixme: assertion #2-3
+            # assertion #4
+            discovered_host = self._assertdiscoveredhost(hostname)
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_discoveredhost.py]: _assertdiscoveredhost method name kind wrong, as we are only trying to get discovered host after making multiple iterations. Yes, we can fix this in some other PR.

File: tests/foreman/cli/test_discoveredhost.py [hunk 3]
```diff
@@ -836,3 +836,63 @@ def test_positive_update_discover_hostname_settings(self):
...
+            2. Ensure the host is provisioned with correct and all the host parameters
+               given during provisioning
+            3. Ensure the entry from discovered host list disappeared
+
+        :CaseImportance: High
+        """
+        param1_key, param1_value = gen_string('alpha'), gen_string('alphanumeric')
+        param2_key, param2_value = gen_string('alpha'), gen_string('alphanumeric')
+        host_params = [
+            '{0}={1}, {2}={3}'.format(param1_key, param1_value, param2_key, param2_value)]
```
Comment (JacobCallahan) [inline tests/foreman/cli/test_discoveredhost.py]: optional: remove the format numbers

File: tests/foreman/cli/test_discoveredhost.py [hunk 4]
```diff
@@ -836,3 +836,63 @@ def test_positive_update_discover_hostname_settings(self):
...
+            # Provision just discovered host
+            DiscoveredHost.provision({
+                'name': discovered_host['name'],
+                'hostgroup': self.configured_env['hostgroup']['name'],
+                'root-password': gen_string('alphanumeric'),
+                'parameters': host_params
+            })
+            # fixme: assertion #5-8
+            provisioned_host = Host.info({
+                'name': '{0}.{1}'.format(
```
Comment (JacobCallahan) [inline tests/foreman/cli/test_discoveredhost.py]: same optional change

File: tests/foreman/cli/test_discoveredhost.py [hunk 5]
```diff
@@ -836,3 +836,63 @@ def test_positive_update_discover_hostname_settings(self):
...
+        param2_key, param2_value = gen_string('alpha'), gen_string('alphanumeric')
+        host_params = [
+            '{0}={1}, {2}={3}'.format(param1_key, param1_value, param2_key, param2_value)]
+        # fixme: assertion #1
+        if not self.configured_env:
+            self.__class__.configured_env = configure_env_for_provision(
+                org=self.org, loc=self.loc)
+        with LibvirtGuest() as pxe_host:
+            hostname = pxe_host.guest_name
+            # fixme: assertion #2-3
```
[the remark this answers is not in the document]
Comment (jyejare) [reply tests/foreman/cli/test_discoveredhost.py]: Ahh. Just forgot remove them... removing now.

File: tests/foreman/cli/test_discoveredhost.py (hunk 1 above)
Comment (jyejare) [reply tests/foreman/cli/test_discoveredhost.py]: Cool .. will update the docstring.

File: tests/foreman/cli/test_discoveredhost.py (hunk 2 above)
Comment (jyejare) [reply tests/foreman/cli/test_discoveredhost.py]: Yes... just reusing this method created 1000 years ago :)

**Comment by jyejare** [comment]
@pgagne @omkarkhatavkar @JacobCallahan , Comments are addressed, please revisit.

**Pull request merged**